### PR TITLE
Issue 3 Fix

### LINKED ISSUE #3: Feature Request: Create snippets while working on another post
If you could make it where I can click a button and a modal pops up and from that I can choose from my code snippets, edit them from there, and create new ones. That would be great.

That way, less tabs, less confusing, better resulting posts in half the time.

### PR DESCRIPTION
**Pull request by JayWood**
Fixes #3 and is really nice to have :)

### Changes
* Add extra dialog structure to allow users to create snippets in-post, fixes [#3](https://github.com/jtsternberg/Code-Snippets-CPT/issues/3)
* Fix rogue text-domain
* Reset form after closing it

### CODE REVIEW

File: lib/CodeSnippitButton.php [hunk 1]
```diff
@@ -14,15 +14,18 @@ function __construct( $cpt, $language ) {
 
 		// Set default programming language taxonomy terms
 		add_action( 'admin_init', array( $this, 'admin_init' ) );
+
+		add_action( 'wp_ajax_insert_snippet', array( $this, 'ajax_insert_snippet' ) );
```
Comment (jtsternberg) [inline lib/CodeSnippitButton.php]: Ajax action names should be prefixed the same way functions are.

File: lib/CodeSnippitButton.php [hunk 2]
```diff
@@ -14,15 +14,18 @@ function __construct( $cpt, $language ) {
...
 	}
 
 	public function button_script() {
 		wp_register_script( $this->script, DWSNIPPET_URL .'/lib/js/'. $this->script .'.js' , array( 'quicktags', 'wpdialogs' ), CodeSnippitInit::VERSION, true );
 		wp_localize_script( $this->script, 'codeSnippetCPT', array(
-			'buttons' => array( 'Cancel' => 'cancel', 'Insert Shortcode' => 'insert' ),
-			'button_img' => DWSNIPPET_URL .'lib/js/icon.png',
-			'button_name' => __( 'Add Snippet', 'code-snippet-cpt' ),
+			'buttons'      => array( 'Cancel' => 'cancel', 'Insert Shortcode' => 'insert' ),
```
Comment (jtsternberg) [inline lib/CodeSnippitButton.php]: The button text needs to be localized, & you should swap the indexes with the values (so the slugs are the indexes).

Comment (jtsternberg) [inline lib/CodeSnippitButton.php]: Ah, I see that is left-over from the previous code. We do need to update that though.

Comment (JayWood) [inline lib/CodeSnippitButton.php]: Haha yea, I was wondering lol.

File: lib/js/code-snippet-button.js [hunk 3]
```diff
@@ -32,20 +45,70 @@ var codeSnippetCPTButton = (function(window, document, $, undefined){
...
+				action    : 'insert_snippet',
+				nonce     : codeSnippetCPT.snippet_nonce,
+				form_data : btn.$.form.serialize(),
+			};
+
+			$.post( ajaxurl, post_data, function( response ){
+				// May be a better way of doing this, but I'm using an anonymous function since I want to
+				// be able to reset the above variables if necessary without globalizing them. - Jay
+				if ( response.success ){
+					console.log( 'made it' );
```
Comment (jtsternberg) [inline lib/js/code-snippet-button.js]: no debug output in PRs. ;)

File: lib/js/code-snippet-button.js [hunk 4]
```diff
@@ -32,20 +45,70 @@ var codeSnippetCPTButton = (function(window, document, $, undefined){
...
+					post_slug = response.data.slug;
+					lang = response.data.language;
+					line_numbers = response.data.line_numbers;
+
+					btn.insert_shortcode( post_slug, line_numbers, lang );
+				} else {
+					// @TODO: show an error or something
```
Comment (jtsternberg) [inline lib/js/code-snippet-button.js]: Let's figure out a plan for the TODOs before merging

**Comment by jtsternberg** [comment]
If you can fix the TODOs, I'll tweak a bit and merge this in. Looks great! If you can add the WP spinner gif during the snippet creating, bonus points.

**Comment by JayWood** [comment]
Works for me, was going to work on that tomorrow anyhow.  As for the debug, must've slipped past me.  I'm used to jshint complaining haha.

**Comment by marbetschar** [comment]
@jtsternberg I've merged this branch into mine, based on your latest master-branch: https://github.com/marbetschar/Code-Snippets-CPT/tree/feature-postsnippet

This pull request looks good to me - despite the fact, the buttons are too close aligned on top:

bildschirmfoto 2016-01-05 um 16 30 30

But this can easily be adjustet by using the following code:

```
#snippet-cpt-form { padding:10px 10px 20px }
```

bildschirmfoto 2016-01-05 um 16 38 06

**Comment by jtsternberg** [comment]
This is in! I'll work on an update to incorporate the ACE editor in the modal as well.

**Pull request merged**